pallets/runtime/src/multisig.rs

```diff
@@ -272,8 +265,50 @@ decl_module! {
                 <NumberOfSigners<T>>::get(&sender) >= sigs_required,
                 Error::<T>::NotEnoughSigners
             );
-            <MultiSigSignsRequired<T>>::insert(&sender, &sigs_required);
-            Self::deposit_event(RawEvent::MultiSigSignaturesRequiredChanged(sender, sigs_required));
+            Self::unsafe_change_sigs_required(sender, sigs_required);
+            Ok(())
+        }
+
+        /// This function allows to replace all existing signers of the given multisig & also change no. of signature required
+        /// NOTE - Once this function get executed no other function of the multisig is allowed to execute until unless
+        /// potential signers accept the authorization and there count should be greater than or equal to the signature required
+        ///
+        /// # Arguments
+        /// * signers - Vector of signers for a given multisig
+        /// * sigs_required - Number of signature required for a given multisig
+        pub fn change_all_signers_and_sigs_required(origin, signers: Vec<Signatory>, sigs_required: u64) -> DispatchResult {
+            let sender = ensure_signed(origin)?;
+            let sender_signer = Signatory::from(AccountKey::try_from(sender.encode())?);
+            ensure!(<MultiSigSignsRequired<T>>::exists(&sender), Error::<T>::NoSuchMultisig);
+            ensure!(signers.len() > 0, Error::<T>::NoSigners);
+            ensure!(u64::try_from(signers.len()).unwrap_or_default() >= sigs_required && sigs_required > 0,
+                Error::<T>::RequiredSignaturesOutOfBounds
+            );
+
+            // Collect the list of all signers present for the given multisig
+            let current_signers = <MultiSigSigners<T>>::iter_prefix(&sender).collect::<Vec<Signatory>>();
+            // Collect all those signers who need to be removed. It means those signers that are not exist in the signers vector
+            // but present in the current_signers vector
+            let old_signers = current_signers.clone().into_iter().filter(|x| !signers.contains(x)).collect::<Vec<Signatory>>();
```

Comment:
clone is not needed

Reply:
Without clone() I can't use the current_signers in the next statement.

Reply:
You can reduce cloning by using
```
let old_signers = current_signers.iter().filter(|&x| !signers.contains(&x)).cloned().collect::<Vec<Signatory>>();
```
This will only clone values that you are keeping in old_signers.
Alternatively, you can make the `current_signers` mutable and use `drain_filter`.